src/constraints.jl

```diff
@@ -0,0 +1,30 @@
+abstract type AbstractConstraint end
+
+struct PopulationConstraint <: AbstractConstraint
+    min_pop::Int
+    max_pop::Int
+end
+
+function PopulationConstraint(graph::BaseGraph,
+                              population_col::AbstractString,
+                              tolerance::Float64=0.01)
+
+    # ideal_pop is a float, because rounding it right now could change the
+    # min/max pop a little bit
+    ideal_pop = graph.total_pop / graph.num_dists
+
+    # no particular reason to not use floor() instead of ceil()
```

Comment:
Wait... but technically, shouldn't we use `floor()` for `max_pop`? That way, `max_pop` is _guaranteed_ to be within the tolerance, yes? (Doesn't matter for large absolute populations, of course.)